From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Eric Kilmer <eric.d.kilmer@gmail.com>
Date: Sun, 12 Feb 2023 13:40:07 -0500
Subject: [PATCH] 4974: gradle: Fix compile classpath for scripts

gradle: Fix compile classpath for scripts sourceSet
gradle: Fix compile classpath for ghidraClass sourceSet
---
 GhidraDocs/build.gradle   | 5 +++++
 gradle/javaProject.gradle | 2 ++
 2 files changed, 7 insertions(+)

diff --git a/GhidraDocs/build.gradle b/GhidraDocs/build.gradle
index 28ab8790e..d484ae3b5 100644
--- a/GhidraDocs/build.gradle
+++ b/GhidraDocs/build.gradle
@@ -21,10 +21,15 @@ sourceSets {
 	ghidraClass {
 		java {
 			srcDir 'GhidraClass/AdvancedDevelopment/contrib/gadc/ghidra_scripts'
+			compileClasspath += main.output
 		}
 	}
 }
 
+configurations {
+   ghidraClassImplementation.extendsFrom implementation
+}
+
 dependencies {
 	api project(':Base') 
 }
diff --git a/gradle/javaProject.gradle b/gradle/javaProject.gradle
index e8564e0a2..422c77a63 100644
--- a/gradle/javaProject.gradle
+++ b/gradle/javaProject.gradle
@@ -123,6 +123,7 @@ sourceSets {
 		java {
 			srcDir 'developer_scripts'
 			srcDir 'ghidra_scripts'
+			compileClasspath += main.output
 		}
 	}
 }
@@ -131,6 +132,7 @@ configurations {
 	integrationTestImplementation.extendsFrom testImplementation
 	integrationTestRuntimeOnly.extendsFrom testRuntimeOnly, integrationTestImplementation
 	pcodeTestImplementation.extendsFrom implementation
+	scriptsImplementation.extendsFrom implementation
 	testArtifacts.extendsFrom testRuntimeOnly
 	integrationTestArtifacts.extendsFrom integrationTestRuntimeOnly
 	screenShotsImplementation.extendsFrom integrationTestImplementation
-- 
2.39.1

